homeassistant/components/unifi/sensor.py

```diff
@@ -279,6 +283,75 @@ def make_wan_latency_entity_description(
     )
 
 
+@callback
+def async_device_temperatures_value_fn(
+    temperature_name: str, hub: UnifiHub, device: Device
+) -> float | None:
+    """Retrieve the temperature of the device."""
+    temperature = _device_temperature(temperature_name=temperature_name, device=device)
+
+    if TYPE_CHECKING:
+        # Checked by async_device_wan_latency_supported_fn
```

Comment:
Copy paste issue.

I still think we can change _device_temperature method in such a way that we don't need this TYPE_CHECKING step

Reply:
> I still think we can change _device_temperature method in such a way that we don't need this TYPE_CHECKING step


Still not 100% sure I understand what you mean here?

Do you mean that `_device_temperature` cannot return `None`?

Reply:
If you can make it always return the expected container fo example

Reply:
If we do that, then the supported_fn need to verify that the container have a `value`

Reply:
I tried to return `{}` but that is not possible. 
Is there another way we can achieve this?

Reply:
You could change the _device_temperature to return the value or None instead, then instead of type checking we could do `return temperature if temperature is not None else 0` or something along those lines

Reply:
If we do that `async_device_temperatures_supported_fn` will always return true. 
I think that is worse?

Reply:
Added something in b11c2b49a474599a40d0336fd74dfdb2ca7e9141